mne/source_estimate.py

```diff
@@ -2639,14 +2639,64 @@ def morph_data_precomputed(subject_from, subject_to, stc_from, vertices_to,
     return stc_to
 
 
+def _spatio_temporal_src_connectivity_vol(src, n_times):
+    assert len(src) == 1  # not a mixed source space
+    from sklearn.feature_extraction import grid_to_graph
+    vertices = np.where(src[0]['inuse'])[0]
+    n_vertices = len(vertices)
+    data = (1 + np.arange(n_vertices))[:, np.newaxis]
+    stc_tmp = VolSourceEstimate(data, vertices, tmin=0., tstep=1.)
+    img = stc_tmp.as_volume(src, mri_resolution=False)
+    img_data = img.get_data()[:, :, :, 0]
+    img_data = img_data.T
+    edges = grid_to_graph(*img_data.shape, mask=(img_data != 0))
+    connectivity = _get_connectivity_from_edges(edges, n_times)
+    return connectivity
+
+
+def _spatio_temporal_src_connectivity_ico(src, n_times):
+    if src[0]['use_tris'] is None:
+        raise RuntimeError("The source space does not appear to be an ico "
+                           "surface. Connectivity cannot be extracted from"
+                           " non-ico source spaces.")
```

Comment:
This implies `oct` also cannot be used, but it probably can. (I know this is from old code but maybe we should fix it here?)

Reply:
I've put an XXX TODO :)